docs/src/main/asciidoc/security-keycloak-authorization.adoc

```diff
@@ -1,39 +1,62 @@
 
-Rather than explicitly enforcing access through specific mechanisms such as role-based access control (RBAC), `quarkus-keycloak-authorization` determines request permissions based on resource attributes such as name, identifier, or Uniform Resource Identifier (URI).
-This process involves sending a `quarkus-oidc`-verified bearer access token to Keycloak Authorization Services for an authorization decision.
+Use the `quarkus-keycloak-authorization` extension only if you are using Keycloak and Keycloak Authorization Services is enabled in your environment to handle authorization decisions.
 
-Use `quarkus-keycloak-authorization` only if you work with Keycloak and have Keycloak Authorization Services enabled to make authorization decisions.
-Use `quarkus-oidc` if you do not work with Keycloak or work with Keycloak but do not have its Keycloak Authorization Services enabled to make authorization decisions.
+If you are not using Keycloak, or if Keycloak is configured without Keycloak Authorization Services, use the `quarkus-oidc` extension instead.
 
-By shifting authorization responsibilities outside your application, you enhance security through various access control methods while eliminating the need for frequent re-deployments whenever security needs evolve.
-In this case, Keycloak acts as a centralized authorization hub, managing your protected resources and their corresponding permissions effectively.
+.How it works
 
-For more information, see the xref:security-oidc-bearer-token-authentication.adoc[OIDC Bearer token authentication] guide.
-It is important to realize that the Bearer token authentication mechanism does the authentication and creates a security identity.
-Meanwhile, the `quarkus-keycloak-authorization` extension applies a Keycloak Authorization Policy to this identity based on the current request path and other policy settings.
+The `quarkus-keycloak-authorization` extension centralizes authorization responsibilities in Keycloak, enhancing security and simplifying application maintenance. The extension:
 
-For more information, see https://www.keycloak.org/docs/latest/authorization_services/index.html#_enforcer_overview[Keycloak Authorization Services documentation].
+1. Uses the `quarkus-oidc` extension to verify bearer tokens.
+2. Sends verified tokens to Keycloak Authorization Services.
+3. Allows Keycloak to evaluate resource-based permissions dynamically, by using attributes such as resource name, identifier, or URI.
+
+By externalizing authorization decisions, you can:
+
+- Implement diverse access control strategies without modifying application code.
+- Reduce redeployment needs as security requirements evolve.
+
+.Compatibility
+
+This extension is compatible only with Quarkus xref:security-oidc-bearer-token-authentication.adoc[OIDC service applications]. It replaces explicit mechanisms such as role-based access control with dynamic authorization policies.
```

Comment:
```suggestion
This extension is compatible only with Quarkus xref:security-oidc-bearer-token-authentication.adoc[OIDC service applications]. It complements explicit mechanisms such as role-based access control with dynamic authorization policies.
```

Reply:
Hi @michalvavrik,

Thank you for your feedback! I’ve applied the suggested change. If you have any additional thoughts or suggestions, please feel free to share—your input is always greatly appreciated.

Just a heads-up: this PR is still a work in progress, so I’ll be refining the existing changes and adding new ones over the coming week. Stay tuned for updates, and thanks again for your support!

Reply:
Hey Rolfe, no, sorry I don't really have a time ATM, I just mentioned your PR and thought that Kc AuthZ shouldn't be saw like replacement for RBAC annotations and HTTP perms etc. I'll try to read your PR when I have a time, but please don't wait for me.